test_programs/compile_success_with_bug/underconstrained_value_detector_5425/src/main.nr

```diff
@@ -0,0 +1,37 @@
+unconstrained fn maximum_price(options: [u32; 3]) -> u32 {
```

Comment:
Please add similar tests for when an unconstrained value in Brillig is:
- A single element of an array with at least two values
- In a nested brillig call, e.g. `foo(bar(xs))` where `xs: [u8; 3]`, `foo` constrains `xs[0]` and `bar` constrains `xs[1]` (i.e. `xs[2]` is unconstrained)
- In a nested type, e.g. `SomeStruct { field_a: [(value, ..), ..], .. }`

Reply:
Can do 3 I believe, for a nested structure. For 1, checking if every element of an array is constrained is for now unsupported as mentioned further below (will make an issue for that).

2 I'll have to make sense of looking at the SSA. Still, won't work with array elements for now in any case.

Reply:
Ok, then please add to the follow up issue to add support for array elements

Reply:
Added a test for briilig calls with nested type results. Mentioned this conversation in https://github.com/noir-lang/noir/issues/6698.